src/main/kotlin/me/odinmain/features/impl/floor7/p3/terminalhandler/TerminalHandler.kt

```diff
@@ -50,6 +51,7 @@ open class TerminalHandler(val type: TerminalTypes, val timeOpened: Long = Syste
     }
 
     fun canClick(slotIndex: Int, button: Int, needed: Int = solution.count { it == slotIndex }): Boolean = when {
+        type == TerminalTypes.MELODY && (TerminalSolver.blockIncorrectClicks || TerminalSolver.renderType == 3) && !TerminalSolver.melodyCorrect -> false
```

Comment:
```suggestion
                type == TerminalTypes.MELODY -> solution.any { it % 9 == 7 }
```
Remove the rest of the changes and test this

Reply:
added as an && to the existing type == Melody; will test

Reply:
tested working